### PR TITLE
fix mod cache not enabling mods correctly

### LINKED ISSUE #320: [Incorrect behavior] Agreeing to enable newly found mods does not properly work
### System Firmware Version

lol

### Atmosphere Version

lmao

### Smash Version

13.0.1

### ARCropolis Version

3.3.0

### Version channel

Stable

### SD card formatting

FAT32

### Can this incorrect behavior be consistently reproduced?

Yes

### Configuration

- [ ] Legacy discovery
- [X] Modern discovery (>= 3.0.0)
- [ ] Irrelevant/Not provided

### Expected behavior

When prompted about new mods being found and asking if they should be enabled, saying yes does not work properly, as it does not take workspaces into account.

### Screenshot

_No response_

### Minimal modpack to reproduce the panic

_No response_

### LINKED ISSUE #341: [Qualify of Life] Don't offer to enable newly found mods if they are already in the current preset
### Explain the improvement(s).

Currently, if you upload a mod while the game is running, enable it from the mod manager, then reboot the game, the cache will still ask you if you'd like to enable the new mod found on the SD. Obviously, you're not *supposed* to add new mods while the game is running, but it seems some savage do just that.

It'd be better to [remove mods in the active preset from the diffed ones](https://github.com/Raytwo/ARCropolis/blob/607a9c02777703ece0428b68924804234d1903bc/src/fs/discover.rs#L135), so that the list only contains the ones that aren't already enabled, preventing the dialog from showing up if the list is empty.

### Provide extra details

Subtract the hashes in the PRESET_HASHES global from new_mods so that you only have the newly found mods that are disabled.

### PR DESCRIPTION
**Pull request by jozz024**
also fixes mods being prompted to be enabled if they were added and enabled while the game was running

closes #341 
closes #320

### CODE REVIEW

File: src/fs/discover.rs [hunk 1]
```diff
@@ -132,16 +132,22 @@ pub fn perform_discovery() -> LaunchPad<StandardLoader> {
...
 
-        let new_mods: HashSet<&Hash40> = new_cache.iter().filter(|cached_mod| !mod_cache.contains(cached_mod)).collect();
+        // Get the workspace name and workspace list
+        let workspace_name: String = storage.get_field("workspace").unwrap_or("Default".to_string());
+        let workspace_list: HashMap<String, String> = storage.get_field_json("workspace_list").unwrap_or_default();
+
+        // Get the preset name from the 
+        let preset_name = &workspace_list[&workspace_name];
+        let mut presets: HashSet<Hash40> = storage.get_field_json(preset_name).unwrap_or_default();
+        let new_mods: HashSet<&Hash40> = new_cache.iter().filter(|cached_mod| !mod_cache.contains(cached_mod)).filter(|cached_mod| !presets.contains(cached_mod)).collect();
```
Comment (Raytwo) [inline src/fs/discover.rs:142]: Could be done in a single filter by using a AND binary operator (&&).
In this case ``!mod_cache.contains(cache_mod) && !presets.contains(cache_mods)`` would only keep the hash if it's not in the cache AND not in the preset.

**Review by Raytwo — CHANGES_REQUESTED** [review]
Almost there! 👌

File: src/fs/discover.rs (hunk 1 above)
Comment (jozz024) [reply src/fs/discover.rs:142]: oh sick, tried to find an and operator online but couldn't find one anywhere

File: src/fs/discover.rs [hunk 2]
```diff
@@ -132,16 +132,22 @@ pub fn perform_discovery() -> LaunchPad<StandardLoader> {
...
             .collect();
 
-        let new_mods: HashSet<&Hash40> = new_cache.iter().filter(|cached_mod| !mod_cache.contains(cached_mod)).collect();
+        // Get the workspace name and workspace list
+        let workspace_name: String = storage.get_field("workspace").unwrap_or("Default".to_string());
+        let workspace_list: HashMap<String, String> = storage.get_field_json("workspace_list").unwrap_or_default();
+
+        // Get the preset name from the 
+        let preset_name = &workspace_list[&workspace_name];
+        let mut presets: HashSet<Hash40> = storage.get_field_json(preset_name).unwrap_or_default();
```
[the remark this answers is not in the document]
Comment (jozz024) [reply src/fs/discover.rs]: resolving due to it not working lol

**Review by Raytwo — APPROVED** [review]
Good job carrying 3.4.0!

**Pull request merged**